### PR TITLE
Clarify behavior of regex conditionals when a no pattern is omitted

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

**Review by danmoseley — APPROVED** [review]

File: docs/standard/base-types/alternation-constructs-in-regular-expressions.md [hunk 1]
```diff
@@ -64,7 +64,7 @@ This language element attempts to match one of two patterns depending on whether
 
 `(?(` *expression* `)` *yes* `|` *no* `)`
 
-where *expression* is the initial pattern to match, *yes* is the pattern to match if *expression* is matched, and *no* is the optional pattern to match if *expression* is not matched. The regular expression engine treats *expression* as a zero-width assertion; that is, the regular expression engine does not advance in the input stream after it evaluates *expression*. Therefore, this construct is equivalent to the following:
+where *expression* is the initial pattern to match, *yes* is the pattern to match if *expression* is matched, and *no* is the optional pattern to match if *expression* is not matched (if a *no* pattern is not provided, it's equivalent to an empty *no*). The regular expression engine treats *expression* as a zero-width assertion; that is, the regular expression engine does not advance in the input stream after it evaluates *expression*. Therefore, this construct is equivalent to the following:
```
Comment (danmoseley) [inline docs/standard/base-types/alternation-constructs-in-regular-expressions.md:67]: This doesn't make clear that you're referring to both `(?(xx)|yy)` and `(?(xx)|yy|)`.
This should probabl yalso be noted here https://github.com/dotnet/docs/blob/main/docs/standard/base-types/regular-expression-language-quick-reference.md#alternation-constructs

Comment (stephentoub) [reply docs/standard/base-types/alternation-constructs-in-regular-expressions.md:67]: I don't understand.  `(?(xx)|yy)` has a no part (`yy`), and `(?(xx)|yy|)` is invalid syntax.

Comment (danmoseley) [reply docs/standard/base-types/alternation-constructs-in-regular-expressions.md:67]: Sorry, fat fingered. `(?(xx)yy)` vs `(?(xx)yy|zz)`

Comment (stephentoub) [reply docs/standard/base-types/alternation-constructs-in-regular-expressions.md:67]: Still not following.  The docs describe the format of the expression:
`(?(` *expression* `)` *yes* `|` *no* `)`
The expression `(?(xx)yy)` lacks a `*no*` part, while `(?(xx)yy|zz)` has a `*no*` part.  Can you elaborate on what would make it more clear?

**Pull request merged**